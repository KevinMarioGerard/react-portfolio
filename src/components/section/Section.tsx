import * as React from 'react';

type SectionHeaderProps = {
    /**
     * Section title
     */
    title: string;

    /**
     * Section header icon
     */
    icon?: string;
};

const SectionHeader: React.FC<SectionHeaderProps> = ({ title, icon = '' }) => (
    <>
        <div className="section__header">
            <i className={`section__header__icon ${icon}`} />
            <span className="section__title">{title.toUpperCase()}</span>
        </div>
        <hr className="separator" />
    </>
);

SectionHeader.displayName = 'Section.Header';

const SectionContent: React.FC = ({ children }) => (
    <div className="section__content">{children}</div>
);

SectionContent.displayName = 'Section.Content';

type SectionProps = {
    /**
     * Section id
     */
    id: string;
};

type Statics = {
    Header: typeof SectionHeader;
    Content: typeof SectionContent;
};

const Section: React.FC<SectionProps> & Statics = ({ id, children }) => (
    <section className="section__container" id={id}>
        {children}
    </section>
);

Section.displayName = 'Section';

Section.Header = SectionHeader;
Section.Content = SectionContent;

export default Section;
