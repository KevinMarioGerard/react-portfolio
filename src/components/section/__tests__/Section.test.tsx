import { render } from '@testing-library/react';
import Section from '../Section';
import * as React from 'react';

describe('Section Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Section id="id1" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<Section id="id1" />);

        expect(container).toMatchSnapshot();
    });
});

describe('Section Header Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Section.Header title="Section Title" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<Section.Header title="Section Title" />);

        expect(container).toMatchSnapshot();
    });
});

describe('Section Content Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Section.Content />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<Section.Content>Mock Section content</Section.Content>);

        expect(container).toMatchSnapshot();
    });
});
