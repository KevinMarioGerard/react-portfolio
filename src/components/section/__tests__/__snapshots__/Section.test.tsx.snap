// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component tests should match snapshot 1`] = `
<div>
  <section
    class="section__container"
    id="id1"
  />
</div>
`;

exports[`Section Content Component tests should match snapshot 1`] = `
<div>
  <div
    class="section__content"
  >
    Mock Section content
  </div>
</div>
`;

exports[`Section Header Component tests should match snapshot 1`] = `
<div>
  <div
    class="section__header"
  >
    <i
      class="section__header__icon "
    />
    <span
      class="section__title"
    >
      SECTION TITLE
    </span>
  </div>
  <hr
    class="separator"
  />
</div>
`;
